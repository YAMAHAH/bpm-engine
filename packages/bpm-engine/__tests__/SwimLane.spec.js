import fs from 'fs';

import BPMEngine from 'bpm-engine';
import History from './Plugins/History';

describe('SwimLane', () => {
  it('Pauses a token and controls continuing it', async (done) => {
    const history = new History();
    const bpm = new BPMEngine({
      plugins: [history],
    });

    // start the process and the flow
    const token = await bpm.createProcessInstance({
      workflowDefinition: fs.readFileSync(`${__dirname}/diagrams/SwimLane.bpmn`, 'utf-8'),
    });
    await token.exec();

    setTimeout(() => {
      expect(history.store).toMatchSnapshot();
      done();
    }, 100);
  });
});
