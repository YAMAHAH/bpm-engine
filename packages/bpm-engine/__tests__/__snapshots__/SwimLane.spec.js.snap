// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwimLane Pauses a token and controls continuing it 1`] = `
Array [
  Object {
    "elementId": "StartEvent_1",
    "lanes": Array [
      Object {
        "id": "Lane_1oypp6e",
        "name": "A",
      },
    ],
    "participant": Object {
      "id": "Participant_1aicb0m",
      "name": "SwimLane",
    },
    "state": "ready",
  },
  Object {
    "elementId": "StartEvent_1",
    "lanes": Array [
      Object {
        "id": "Lane_1oypp6e",
        "name": "A",
      },
    ],
    "participant": Object {
      "id": "Participant_1aicb0m",
      "name": "SwimLane",
    },
    "state": "active",
  },
  Object {
    "elementId": "StartEvent_1",
    "lanes": Array [
      Object {
        "id": "Lane_1oypp6e",
        "name": "A",
      },
    ],
    "participant": Object {
      "id": "Participant_1aicb0m",
      "name": "SwimLane",
    },
    "state": "complete",
  },
  Object {
    "elementId": "SequenceFlow_02qqmbt",
    "lanes": undefined,
    "participant": Object {
      "id": "Participant_1aicb0m",
      "name": "SwimLane",
    },
    "state": "ready",
  },
  Object {
    "elementId": "SequenceFlow_02qqmbt",
    "lanes": undefined,
    "participant": Object {
      "id": "Participant_1aicb0m",
      "name": "SwimLane",
    },
    "state": "active",
  },
  Object {
    "elementId": "SequenceFlow_02qqmbt",
    "lanes": undefined,
    "participant": Object {
      "id": "Participant_1aicb0m",
      "name": "SwimLane",
    },
    "state": "complete",
  },
  Object {
    "elementId": "Task_0qvmu3d",
    "lanes": Array [
      Object {
        "id": "Lane_1kmmx7p",
        "name": "B",
      },
      Object {
        "id": "Lane_0n7eabp",
        "name": "B2",
      },
    ],
    "participant": Object {
      "id": "Participant_1aicb0m",
      "name": "SwimLane",
    },
    "state": "ready",
  },
  Object {
    "elementId": "Task_0qvmu3d",
    "lanes": Array [
      Object {
        "id": "Lane_1kmmx7p",
        "name": "B",
      },
      Object {
        "id": "Lane_0n7eabp",
        "name": "B2",
      },
    ],
    "participant": Object {
      "id": "Participant_1aicb0m",
      "name": "SwimLane",
    },
    "state": "active",
  },
  Object {
    "elementId": "Task_0qvmu3d",
    "lanes": Array [
      Object {
        "id": "Lane_1kmmx7p",
        "name": "B",
      },
      Object {
        "id": "Lane_0n7eabp",
        "name": "B2",
      },
    ],
    "participant": Object {
      "id": "Participant_1aicb0m",
      "name": "SwimLane",
    },
    "state": "complete",
  },
  Object {
    "elementId": "SequenceFlow_03yb0xa",
    "lanes": undefined,
    "participant": Object {
      "id": "Participant_1aicb0m",
      "name": "SwimLane",
    },
    "state": "ready",
  },
  Object {
    "elementId": "SequenceFlow_03yb0xa",
    "lanes": undefined,
    "participant": Object {
      "id": "Participant_1aicb0m",
      "name": "SwimLane",
    },
    "state": "active",
  },
  Object {
    "elementId": "SequenceFlow_03yb0xa",
    "lanes": undefined,
    "participant": Object {
      "id": "Participant_1aicb0m",
      "name": "SwimLane",
    },
    "state": "complete",
  },
  Object {
    "elementId": "Task_0qjan9z",
    "lanes": Array [
      Object {
        "id": "Lane_1kmmx7p",
        "name": "B",
      },
      Object {
        "id": "Lane_0x63534",
        "name": "B1",
      },
    ],
    "participant": Object {
      "id": "Participant_1aicb0m",
      "name": "SwimLane",
    },
    "state": "ready",
  },
  Object {
    "elementId": "Task_0qjan9z",
    "lanes": Array [
      Object {
        "id": "Lane_1kmmx7p",
        "name": "B",
      },
      Object {
        "id": "Lane_0x63534",
        "name": "B1",
      },
    ],
    "participant": Object {
      "id": "Participant_1aicb0m",
      "name": "SwimLane",
    },
    "state": "active",
  },
  Object {
    "elementId": "Task_0qjan9z",
    "lanes": Array [
      Object {
        "id": "Lane_1kmmx7p",
        "name": "B",
      },
      Object {
        "id": "Lane_0x63534",
        "name": "B1",
      },
    ],
    "participant": Object {
      "id": "Participant_1aicb0m",
      "name": "SwimLane",
    },
    "state": "complete",
  },
  Object {
    "elementId": "SequenceFlow_0eeydo7",
    "lanes": undefined,
    "participant": Object {
      "id": "Participant_1aicb0m",
      "name": "SwimLane",
    },
    "state": "ready",
  },
  Object {
    "elementId": "SequenceFlow_0eeydo7",
    "lanes": undefined,
    "participant": Object {
      "id": "Participant_1aicb0m",
      "name": "SwimLane",
    },
    "state": "active",
  },
  Object {
    "elementId": "SequenceFlow_0eeydo7",
    "lanes": undefined,
    "participant": Object {
      "id": "Participant_1aicb0m",
      "name": "SwimLane",
    },
    "state": "complete",
  },
  Object {
    "elementId": "EndEvent_1vv0vt4",
    "lanes": Array [
      Object {
        "id": "Lane_1oypp6e",
        "name": "A",
      },
    ],
    "participant": Object {
      "id": "Participant_1aicb0m",
      "name": "SwimLane",
    },
    "state": "ready",
  },
  Object {
    "elementId": "EndEvent_1vv0vt4",
    "lanes": Array [
      Object {
        "id": "Lane_1oypp6e",
        "name": "A",
      },
    ],
    "participant": Object {
      "id": "Participant_1aicb0m",
      "name": "SwimLane",
    },
    "state": "active",
  },
  Object {
    "elementId": "EndEvent_1vv0vt4",
    "lanes": Array [
      Object {
        "id": "Lane_1oypp6e",
        "name": "A",
      },
    ],
    "participant": Object {
      "id": "Participant_1aicb0m",
      "name": "SwimLane",
    },
    "state": "complete",
  },
]
`;
